ICAFIX/scripts/run_icasso.m

```diff
@@ -0,0 +1,247 @@
+% afterwards. Most but not all of MELODIC's outputs are created.
+% 
+% All of this functions inputs are strings
+%
+% Required Inputs (variable names verbatim from hcp_fix_multi_run.sh):
+%   Dim          : Data diminsionaliy estimate, typically Wishart-based
+%   concatfmri   : File name of original 4d time series without extension
+%   concatfmrihp : File name of high-passed 4d time series without extension
+%   ConcatFolder : Directory which contains concatfmri
+%   tr           : inverse sampling frequency of fMRI timeseries 
+%
+% Optional Inputs:
+%   vis     : Whether to create and save icasso figures, see icasso.m 'basic' (default) or 'off'
+%   nICA    : Number of ICA repetitions per icasso repetition, @ delimited string, (default = '100')
+%   nClust  : Number of icasso repetitions (default = '1')
+%   maxIter : Maximum number of iterations per ica fit (default = '1000')
+%
+% example inputs:
+% Dim = '41';
+% concatfmri = 'tfMRI_EMOTION_RL_LR';
+% concatfmrihp = 'tfMRI_EMOTION_RL_LR_hp0';
+% ConcatFolder = '/mnt/myelin/burke/HCPpipelines/dev_study/100307/MNINonLinear/Results/tfMRI_EMOTION_RL_LR_ICASSO';
+% tr = '0.72';
+% vis = 'basic';
+% nICA = '2@2';
+% nClust = '2';
+% maxIter = '100';
+
+% Created 2024-06-05
+% Burke Rosen
+
+% ToDo:
+% () Currently the vnts file and brainMaskFile paths and names are inferred from
+%    ConcatFolder, concatfmrihp, and concatfmri. Maybe they should be their own 
+%    arguments for flexibility.
+% () Spin off a stand alone melodic mixture modeling only matlab wrapper utility
+
+%% parse parameters
+if nargin < 5 || any(cellfun(@isempty,{Dim,concatfmri,concatfmrihp,ConcatFolder}))
+  error('Dim, concatfmri, concatfmrihp, ConcatFolder, and tr are required!')
+end
+Dim = str2double(Dim);
+tr = str2double(tr);
+if nargin < 6 || isempty(vis) 
+  vis = 'basic';
+end
+if ~ismember(vis,{'basic','off'})
+  warning('vis must be ''basic'' or ''off'', reverting to ''basic''')
+  vis = 'basic';
+end
+if nargin < 7 || isempty(nICA) 
+  nICA = 100;
+else
+  nICA = cellfun(@str2double,regexp(nICA,'@','split'));
+end
+if nargin < 8 || isempty(nClust) 
+  nClust = 1;
+else
+  nClust = str2double(nClust);
+end
+if numel(nICA) ~= nClust
+  error('nICA must be a @ delimited list with a number of elements equal to nClust')
+end
+if nargin < 9 || isempty(maxIter) 
+  maxIter = 1000;
+else
+  maxIter = str2double(maxIter);
+end
+
+%% parse paths
+% inputs
+vntsFile = sprintf('%s/%s_vnts.nii.gz',ConcatFolder,concatfmrihp);
+brainMaskFile = sprintf('%s/%s_brain_mask.nii.gz',ConcatFolder,concatfmri);
+if ~exist(vntsFile,'file');error('%s doesn''t exist!',vntsFile);end
+if ~exist(brainMaskFile,'file');error('%s doesn''t exist!',brainMaskFile);end
+
+% outputs
+outDir = sprintf('%s/%s.ica/filtered_func_data.ica',ConcatFolder,concatfmrihp);
+[~,~] = unix(['mkdir -p ' outDir]);
+
+%% load data, reshape to 2d, and apply brain mask
+vnts = double(niftiread(vntsFile));
```

Comment:
niftiread is not a function I am familiar with.  I typically use FSL's read_avw and save_avw to avoid adding dependencies.

Reply:
I didn't know that.  Perhaps could check if the function is present and fall back to read_avw/save_avw if not?